src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,636 @@
+        
+        try {
+            AmazonBillingCostCenterDTO costCenterDTO = convertCostCenterToAmazonBillingCostCenterDTO(amazonAccountDetail.getCostCenter());
+            updateTransactionDTOChartAccountSubAccount(transactionDTO, costCenterDTO, amazonAccountDetail.getKfsAccount());
+            updateTransactionDTOObjectCodes(transactionDTO, costCenterDTO);
+            updateTransactionDTOProjectCodeNumber(transactionDTO, costCenterDTO);
+            transactionDTO.setOrganizationReferenceId(costCenterDTO.getOrgReferenceId());
+        } catch (Exception e) {
+            transactionDTO.setTransactionInputError(true);
+            LOG.error("buildAmazonDistributionOfIncomeTransactionDTO: There was a problem parsing cost center.", e);
+        }
+        return transactionDTO;
+    }
+    
+    protected KualiDecimal convertCostStringToKualiDecimal(String costString) {
+        KualiDecimal cost = KualiDecimal.ZERO;
+        try {
+            double costDouble = Double.valueOf(costString);
+            final double oneCent = .01;
+            if (costDouble >= oneCent) {
+                cost = new KualiDecimal(costDouble);
+            }
+        } catch (Throwable e) {
+            LOG.error("convertCostStringToKualiDecimal() Unable to convert " + costString + " to a double.", e); 
+        }
+        return cost;
+    }
+    
+    protected AmazonBillingCostCenterDTO convertCostCenterToAmazonBillingCostCenterDTO(String costCenter) throws Exception {
+        AmazonBillingCostCenterDTO costCenterDTO = new AmazonBillingCostCenterDTO();
+        
+        if (StringUtils.contains(costCenter, KFSConstants.DASH)) {
+            String[] costCenterArray = StringUtils.splitPreserveAllTokens(costCenter, KFSConstants.DASH);
+            if (costCenterArray.length == 2) {
+                costCenterDTO.setAccountNumber(costCenterArray[0]);
+                costCenterDTO.setSubAccountNumber(costCenterArray[1]);
+            } else if (costCenterArray.length == 7) {
+                costCenterDTO.setChartCode(costCenterArray[0]);
+                costCenterDTO.setAccountNumber(costCenterArray[1]);
+                costCenterDTO.setSubAccountNumber(costCenterArray[2]);
+                costCenterDTO.setObjectCode(costCenterArray[3]);
+                costCenterDTO.setSubObjectCode(costCenterArray[4]);
+                costCenterDTO.setProjectCode(costCenterArray[5]);
+                costCenterDTO.setOrgReferenceId(costCenterArray[6]);
+            } else {
+                throw new Exception("There was an incorrect number of elements in the Cost Center");
+            }
+        } else {
+            costCenterDTO.setAccountNumber(costCenter);
+        }
+        
+        return costCenterDTO;
+    }
+    
+    private void updateTransactionDTOChartAccountSubAccount(AmazonBillingDistributionOfIncomeTransactionDTO transactionDTO, AmazonBillingCostCenterDTO costCenterDTO,
+            String amazonDetailKFSAccount) {
+        Account account;
```

Comment:
seems like the first part of this method could be extracted into a retrieve account method